syntax = "proto3";

option go_package = "capstone-2025-30/proto";

enum EError {
  EE_UNSPECIFIED = 0;

  // 공통
  EE_API_FAILED = 10000;
  EE_API_AUTH_FAILED = 10001;
  EE_INVALID_ARGUMENT = 10002;
  EE_DB_OPERATION_FAILED = 10003;

  // store
  EE_STORE_ALREADY_EXISTS = 20000;
  EE_STORE_NOT_FOUND = 20001;
  EE_STORE_UPDATE_NO_FIELDS = 20002;

  // inquiry
  EE_INQUIRY_STREAM_FAILED = 30000;

  // order
  EE_ORDER_AND_NOTIFICATION_AND_MESSAGE_DB_ADD_FAILED = 40000;
  EE_ORDER_NOT_FOUND = 40001;

  // menu
  EE_MENU_ALREADY_EXISTS = 50000;
  EE_MENU_NOT_FOUND = 50001;
  EE_MENU_CATEGORY_NOT_FOUND = 50002;

  // message
  EE_NOTIFICATION_NOT_FOUND = 60000;
  EE_MESSAGE_CREATE_FAILED = 60001;
}